### ISSUE TITLE
Property headerline cannot be set back to NULL/Empty

### ISSUE THREAD

**Issue by dstidham63** [simoc/csvjdbc#52]
Once a header string value has been supplied to headerline, the property cannot be set to NULL/empty by deleting the header string.  It behaves as if there is a single header column instead of no setting.

For example, if a file contains 10 columns, and a header string of 10 values is supplied, all works as expected.  If you delete the string, now the file read throws an exception that 10 columns read, when 1 was expected.  Setting the ignoreNonParseable to TRUE simply doesn't read any data.

charset = UTF-8
fileExtension = .txt
quotechar = "
separator = |
skipLeadingLines = 1
suppressHeaders = true

**Comment by simoc** [comment]
The database connection properties cannot be changed after connecting to a database.

If you want to use different database connection properties for some SQL queries, then make a second database connection
with different database connection properties. Something like:

```
    Properties props = new Properties();
    props.put("headerline", "ID,NAME");
    props.put("suppressHeaders", "true");
    Connection conn1 = DriverManager.getConnection("jdbc:relique:csv:c:/temp", props);
    ResultSet results1 = conn1.createStatement().executeQuery("SELECT * FROM sample1");
    processResults(results1);
    
    props.put("headerline", "");
    props.put("suppressHeaders", "false");
    Connection conn2 = DriverManager.getConnection("jdbc:relique:csv:c:/temp", props);
    ResultSet results2 = conn2.createStatement().executeQuery("SELECT * FROM sample2");
    processResults(results2);
```

**Comment by simoc** [comment]
I also discovered that the database connection property `headerline` works correctly when it is set to a list of column names, or is not set in the properties.

However, if I set this connection property to an an empty string (`""`) then I always get an `Wrong number of columns in line: 1 Columns read: 10 Columns expected: 1` error.

With the following simple CSV file `sample2.csv`

    1,Auckland,New Zealand,NZ,100,999,true,1,0,0
    2,Wellington,New Zealand,NZ,200,999,true,1,0,0

The following source code works correctly without the `headerline` property but reports an errors if this property is set to an empty string. That is a bug in CsvJdbc.

```
    Properties props = new Properties();
    props.put("suppressHeaders", "true");
    props.put("headerline", ""); // works correctly without this line
    Connection conn1 = DriverManager.getConnection("jdbc:relique:csv:c:/temp", props);
    ResultSet results1 = conn1.createStatement().executeQuery("SELECT * FROM sample2");
```

**Comment by dstidham63** [comment]
Thank you for the details.  I am using the driver via DBeaver CE and have no control over how it is being called from that tool.  When changes to the driver's properties are made after the connection is in place, DBeaver closes the connection and re-opens it, refreshing the properties being sent as it initializes the connection.  I suspect that it is doing exactly as you have tested, sending an equivalent setting _props.put("headerline", "");_

Is this behavior something that can be addressed in the driver itself, or would you recommend I reach out the DBeaver CE team to open a request that they investigate this from their side?

**Comment by simoc** [comment]
I have fixed this problem in the CsvJdbc driver and I will create a new CsvJdbc version with this fix included.

**Comment by simoc** [comment]
This issue is fixed in CsvJdbc version 1.0.39.
